massa-deferred-calls/src/lib.rs

```diff
@@ -0,0 +1,482 @@
+use call::{DeferredCallDeserializer, DeferredCallSerializer};
+use macros::DEFERRED_CALL_TOTAL_GAS;
+use massa_db_exports::{
+    DBBatch, ShareableMassaDBController, CRUD_ERROR, DEFERRED_CALLS_PREFIX,
+    DEFERRED_CALL_DESER_ERROR, DEFERRED_CALL_SER_ERROR, KEY_DESER_ERROR, STATE_CF,
+};
+use massa_serialization::{DeserializeError, Deserializer, Serializer};
+use registry_changes::{
+    DeferredRegistryChanges, DeferredRegistryChangesDeserializer, DeferredRegistryChangesSerializer,
+};
+
+/// This module implements a new version of the Autonomous Smart Contracts. (ASC)
+/// This new version allow asynchronous calls to be registered for a specific slot and ensure his execution.
+mod call;
+pub mod registry_changes;
+pub mod slot_changes;
+
+#[cfg(test)]
+mod tests;
+
+#[macro_use]
+mod macros;
+
+pub use call::DeferredCall;
+use massa_ledger_exports::{SetOrDelete, SetOrKeep};
+use massa_models::{
+    amount::Amount,
+    config::{DEFERRED_CALL_MAX_POOL_CHANGES, MAX_ASYNC_GAS, THREAD_COUNT},
```

Comment:
do not use constants, pass a config structure

Reply:
done